### PR TITLE
fix: Ensure to fallback to old editor properly on 25

### LINKED ISSUE #982: Newly integrated rich text editor does not load
As the title says, if I switch to the newly integrated text editor for editing my notes, nothing happens and only a loading screen is shown when I want to edit a note. Moreover, there's an entry in the logs as well as my console which might be related.

## Steps to reproduce
1. Switch to new text editor
2. Try to edit a note.

### Expected behaviour
Text editor opens and note can be edited

### Actual behaviour
Infinite loading screen

### Screenshots
afbeelding

## Server
Please complete the following information.

- Notes app version: 4.7.0
- Nextcloud version: 25.0.4
- OS: Fedora Linux 37
- Web server: Apache/2.4.56 (Fedora Linux)
- PHP version: PHP 8.1.17 (cli) (built: Mar 14 2023 19:50:20) (NTS gcc x86_64)
- Database: mariadb  Ver 15.1 Distrib 10.5.18-MariaDB, for Linux (x86_64) using  EditLine wrapper

**Nextcloud configuration:**

<details>

```

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.joshuanoeske.de",
            "127.0.0.1"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "25.0.4.1",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "default_language": "de",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "enabledPreviewProviders": [
            "OC\\Preview\\BMP",
            "OC\\Preview\\GIF",
            "OC\\Preview\\JPEG",
            "OC\\Preview\\MarkDown",
            "OC\\Preview\\MP3",
            "OC\\Preview\\PNG",
            "OC\\Preview\\TXT",
            "OC\\Preview\\XBitmap",
            "OC\\Preview\\OpenDocument",
            "OC\\Preview\\Krita",
            "OC\\Preview\\HEIC",
            "OC\\Preview\\TIFF",
            "OC\\Preview\\Movie",
            "OC\\Preview\\MKV",
            "OC\\Preview\\MP4",
            "OC\\Preview\\AVI"
        ],
        "default_phone_region": "DE",
        "installed": true,
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": "0"
        },
        "htaccess.RewriteBase": "\/",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "theme": "",
        "loglevel": 1,
        "logfile": "\/var\/log\/nextcloud\/nextcloud.log",
        "logfilemode": 384,
        "lost_password_link": "disabled",
        "updater.release.channel": "stable",
        "twofactor_enforced": "true",
        "twofactor_enforced_groups": [],
        "twofactor_enforced_excluded_groups": [],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "memories.gis_type": 1
    }
}

```
</details>

## Client
Please complete the following information.

- Browser (incl. version): Mozilla Firefox 110.0
 - OS: Fedora Linux 37

## Log files
Add relevant parts of your `nextcloud.log` and/or your browser's JavaScript console here.
```
{"reqId":"ZBmQaHDvVe1M3823b0LLkAABigc","level":2,"time":"2023-03-21T11:09:28+00:00","remoteAddr":"79.194.174.116","user":"joshua.noeske","app":"no app in context","method":"GET","url":"/apps/notes/note/204730","message":"Invalid config data provided to provideInitialState by notes","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:109.0) Gecko/20100101 Firefox/110.0","version":"25.0.4.1","data":[],"id":"6419915547d69"}
```

And from my browser:
```
Uncaught (in promise) TypeError: window.OCA.Text is undefined
    t NoteRich.vue:86
    c Welcome.vue:20
    v Welcome.vue:20
    F Welcome.vue:20
    ti Welcome.vue:20
    o Welcome.vue:20
NoteRich.vue:86
```

### PR DESCRIPTION
**Pull request by juliushaertl**
Fixes #982 

I was missing the case where the text is enabled but still on 25 meaning that createEditor is not available.

### CODE REVIEW

File: src/components/AppSettings.vue [hunk 1]
```diff
@@ -83,6 +83,9 @@ export default {
 	},
 
 	created() {
+		if (!window?.OCA?.Text?.createEditor) {
```
Comment (mejo-) [inline src/components/AppSettings.vue:86]:
```suggestion
		if (!window.OCA.Text?.createEditor) {
```

same here

File: src/App.vue [hunk 2]
```diff
@@ -101,7 +101,7 @@ export default {
 			deletedNotes: [],
 			refreshTimer: null,
 			helpVisible: false,
-			editorHint: loadState('notes', 'editorHint', '') === 'yes' && window.oc_appswebroots.text,
+			editorHint: loadState('notes', 'editorHint', '') === 'yes' && window?.OCA?.Text?.createEditor,
```
Comment (mejo-) [inline src/App.vue:104]:
```suggestion
			editorHint: loadState('notes', 'editorHint', '') === 'yes' && window.OCA.Text?.createEditor,
```

`window.OCA` should always be there, no?

**Review by mejo- — APPROVED** [review]
Just minor comments, otherwise code looks good to me. Didn't test though.

File: src/App.vue (hunk 2 above)
Comment (juliushaertl) [reply src/App.vue]: Right, adjusted that :)

**Pull request merged**